optaplanner-core/src/test/java/org/optaplanner/core/api/score/stream/ConstraintCollectorsTest.java

```diff
@@ -1324,13 +1325,229 @@ public void minNotComparable() {
         assertResult(collector, container, null);
     }
 
+    @Test
+    public void minComparableMapped() {
```

Comment:
This test smells like its "abstracted". (I am not saying it is an abstracted test)
I personally favor concrete tests a lot of abstracted tests.

For example:

```
// Concrete test
assertEquals(7, calculator.sum(2, 5));
assertEquals(7, calculator.sum(4, 3));
assertEquals(9, calculator.sum(4, 5));

// Abstracted test
for (i = 0; i < 7; i++) {
  for (j = 0; j < 7; i++) {
   assertEquals(i + j, calculator.sum(i, j));
  }
}
```

Abstracted tests do more tests, but often don't increase test coverage. They make it hard to read what happens in corner cases - which can lead to invalid tests. And they often copy-paste the mistakes from the unit into the unit test code (especially corner cases).